lib/strings.nix

```
@@ -205,7 +207,20 @@ rec {
        normalizePath "/a//b///c/"
        => "/a/b/c/"
   */
-  normalizePath = s: (builtins.foldl' (x: y: if y == "/" && hasSuffix "/" x then x else x+y) "" (stringToCharacters s));
+  normalizePath = s:
+    warnIf
+      (isPath s)
+      ''
+        lib.strings.normalizePath: The argument (${toString s}) is a path value, but only strings are supported.
+            Path values are always normalised in Nix, so there's no need to call this function on them.
+            This function also imports the path into the Nix store and returns the store path, the same as "''${path}" will, which may not be desired.
```

```suggestion
            This function also copied the path to the Nix store and returns the store path, the same as "''${path}" will, which may not be what you want.
```

Import is a loaded term around here, and may produce the wrong associations. Let's stick to more everyday words.

Desired - by whom? Passive voice sometimes obscures what you may want to say. What is the desired message here?

---

Applied :+1: